cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -418,6 +418,11 @@ func (p *BicepProvider) deploymentState(
 		return nil, fmt.Errorf("deployment state error: %w", err)
 	}
 
+	// state is valid if the last deployment failed
+	if *prevDeploymentResult.Properties.ProvisioningState == armresources.ProvisioningStateFailed {
```

Comment:
This is currently safe because we rely on `latestDeploymentResult` which relies on `findCompletedDeployments` which filters to only `Failed` and `Succeeded`.

Would it be safer still to negate the check for `!= Succeeded`? I could see other states being added in the future that is considered "completed" but not "succeeded", i.e. Cancelled. And cache miss is better than invalid cache usage.